### PR TITLE
man: use MIT-0 license for example codes in daemon(7)

### PR DESCRIPTION
**Pull request by yuwata**
Reapplies systemd/systemd#35359

### CODE REVIEW

**Comment by bluca** [comment]
@mgorny @poettering @keszybz @jengelh @vmcj
you have authored some changes to the autotools snippets contained in this paragraph of our docs: https://www.freedesktop.org/software/systemd/man/latest/daemon.html#Installing%20systemd%20Service%20Files

The default license of the repository is LGPL-2.1+. For documentation snippets, in order to facilitate users just lifting them as-is from the documentation and using them in their projects, to integrate with systemd, we prefer to use MIT-0, which is a no-strings-attached-license, in layman's terms similar to public domain (without actually being public domain, because the meaning of that depends on local legislation).

Are you ok with such a change? If so, please click on the 'approve' button or leave a comment saying so. Thanks!

**Comment by yuwata** [comment]
Only four authors (except for @keszybz and @poettering) touched the example codes 
c91d636fadc @vmcj
fc9acf252ff @jengelh
5486855f7d9 @mgorny
73da50222cd @osmith42

Could you confirm the license change?

**Comment by mgorny** [comment]
I'm okay with changing the license.

**Comment by keszybz** [comment]
The changes from @vmcj and @osmith42 are simple syntax changes that are not copyrightable on their own. (Those are still good changes and useful, but we just don't need to consider them for the purposes of relicensing.)   All the remaining people who touched this (fairly trivial) snippet have given assent.

**Comment by keszybz** [comment]
On Tue, Nov 26, 2024 at 03:22:00AM -0800, Jan Engelhardt wrote:
> What about the rest of the documentation, does anyone know if there was ever a proposition to move it to e.g. CC-BY-SA?

It's very useful for us that almost all of codebase is under the same
license. This allows us to move bits of code without having to worry
about adjusting licensing or getting acceptance from previous authors,
etc. This also includes the docs: we take snippets of real code and
put them in the docs, or comments from the code, or sometimes move
things the other way.

Also, changing the license of the docs would entail getting acceptance
from up to 833 people. (This is the count of people who touched
man/ and docs/. Some contributions are probably trivial so the number
would go down, but it'd still be a lot.)

**Comment by bluca** [comment]
> The changes from @vmcj and @osmith42 are simple syntax changes that are not copyrightable on their own. (Those are still good changes and useful, but we just don't need to consider them for the purposes of relicensing.)

That may very well be, but in order to avoid creating any perception issues or doubts, it is better when dealing with license changes to err on the side of getting too many approvals, rather than too few. Perception of the licensing status of a project is what really matters in 99.999% of cases anyway.

> All the remaining people who touched this (fairly trivial) snippet have given assent.

@poettering hasn't, yet, I'm sure he will, but it needs to be recorded too

**Comment by keszybz** [comment]
Excluding people who only made trivial changes to pages is standard practice. This is what I have seen done in other projects which did license changes, and this is also what we've done in the past. I.e. people who did formatting fixes or mass syntax updates or spelling fixes and so on are filtered out from the list of people who have to give assent. This is for purely practical reasons — each additional person makes the process take a bit longer. In this case there wasn't any doubt about the level of changes, you linked to the patches yourself.

**Comment by poettering** [comment]
yeah, fine by me

**Comment by vmcj** [comment]
Fine by me.
